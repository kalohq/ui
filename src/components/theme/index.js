/* @flow */
export {default} from './theme';
