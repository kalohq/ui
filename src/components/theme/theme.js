/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
