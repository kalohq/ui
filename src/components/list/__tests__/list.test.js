/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';

import List from '../list';
import Paper from '../../paper';

describe('components/list', () => {
  const create = (spread = {}) => {
    const props = {spaced: 'medium', ...spread};
    const element = renderer
      .create(
        <ThemeProvider theme={theme}>
          <List {...props}>
            <Paper>List Item One</Paper>
            <Paper>List Item Two</Paper>
            <Paper>List Item Three</Paper>
            <Paper>List Item Four</Paper>
          </List>
        </ThemeProvider>
      )
      .toJSON();

    return {props, element};
  };

  it('should shallow render as expected', () => {
    const {element} = create();
    const result = shallow(element);
    expect(result).toMatchSnapshot();
  });

  it('should render as a grid', () => {
    const {element} = create();
    const result = shallow(element);
    expect(result).toMatchSnapshot();
  });
});
