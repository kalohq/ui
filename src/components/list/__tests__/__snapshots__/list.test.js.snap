// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/list should render as a grid 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="e13caurm0 css-z4vh1z"
    style={Object {}}
    wrap={undefined}
  >
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item One
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Two
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Three
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Four
      </div>
    </li>
  </ul>,
  "nodes": Array [
    <ul
      className="e13caurm0 css-z4vh1z"
      style={Object {}}
      wrap={undefined}
    >
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item One
        </div>
      </li>
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item Two
        </div>
      </li>
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item Three
        </div>
      </li>
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item Four
        </div>
      </li>
    </ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ul
    className="e13caurm0 css-z4vh1z"
    style={Object {}}
    wrap={undefined}
  >
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item One
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Two
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Three
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Four
      </div>
    </li>
  </ul>,
}
`;

exports[`components/list should shallow render as expected 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <ul
    className="e13caurm0 css-z4vh1z"
    style={Object {}}
    wrap={undefined}
  >
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item One
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Two
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Three
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Four
      </div>
    </li>
  </ul>,
  "nodes": Array [
    <ul
      className="e13caurm0 css-z4vh1z"
      style={Object {}}
      wrap={undefined}
    >
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item One
        </div>
      </li>
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item Two
        </div>
      </li>
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item Three
        </div>
      </li>
      <li
        className="e13caurm1 css-1snizm0"
        style={Object {}}
      >
        <div
          className="e4o9m4l0 css-11l6ekn"
          onClick={undefined}
          style={Object {}}
        >
          List Item Four
        </div>
      </li>
    </ul>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": null,
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ul
    className="e13caurm0 css-z4vh1z"
    style={Object {}}
    wrap={undefined}
  >
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item One
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Two
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Three
      </div>
    </li>
    <li
      className="e13caurm1 css-1snizm0"
      style={Object {}}
    >
      <div
        className="e4o9m4l0 css-11l6ekn"
        onClick={undefined}
        style={Object {}}
      >
        List Item Four
      </div>
    </li>
  </ul>,
}
`;
